tests/functional/specs/spec_with_hyphenated_hiddenimport.spec

```diff
@@ -0,0 +1,29 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2023, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+#-----------------------------------------------------------------------------
+
+# Verify that a hyphenated module is a valid hiddenimport (see # issue #8591).
+
+app_name = "spec_with_hyphenated_hiddenimport"
+
+a = Analysis(['several-scripts/main-script1.py'],
+             hiddenimports=['fake-hyphenated-module.py'])
```

Comment:
Is this test case, `fake-hyphenated-module.py` legit? I.e., we are talking about `py` submodule of `fake-hyphenated-module` package and not a filename?

Reply:
Oh, yea, I suppose it should be `fake-hyphenated-module` without the `.py`. I don't think it affects the coverage of the bug, but it indeed makes more sense without the misleading "suffix"